# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/GlobalObject.cpp
# Commit: e9c2f256240a
# Full Hash: e9c2f256240a1f9bb09b10ac641bbd141eedb809
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-03-06 09:33:16
# Regressor Bug: 1689413
# File Overlap Count: 1
# Description:
#   Bug 1689413 part 2 - Remove owned BaseShapes. r=jonco
#   
#   At this point all BaseShapes are unowned (= immutable, shared) so we can remove the
#   UnownedBaseShape type and the `BaseShape::unowned_` field.
#   
# ==============================================================================

diff -r 1ae4c30bb931 -r e9c2f256240a js/src/vm/GlobalObject.cpp
--- a/js/src/vm/GlobalObject.cpp	Sat Mar 06 01:05:19 2021 +0000
+++ b/js/src/vm/GlobalObject.cpp	Sat Mar 06 01:05:19 2021 +0000
@@ -1079,7 +1079,7 @@
 
   uint32_t slot = holder->slotSpan();
   RootedShape last(cx, holder->lastProperty());
-  Rooted<UnownedBaseShape*> base(cx, last->base()->unowned());
+  Rooted<BaseShape*> base(cx, last->base());
 
   RootedId id(cx, NameToId(name));
   Rooted<StackShape> child(cx, StackShape(base, id, slot, 0));